packages/server/api/src/app/flows/issues/issues-service.ts

```diff
@@ -1,49 +1,113 @@
-import { Issue, IssueStatus, ListIssuesParams, PopulatedIssue } from '@activepieces/ee-shared'
 import { rejectedPromiseHandler } from '@activepieces/server-shared'
-import { ActivepiecesError, ApId, apId, ErrorCode, isNil, SeekPage, spreadIfDefined, TelemetryEventName } from '@activepieces/shared'
+import { ActivepiecesError, apId, ApId, ErrorCode, ExecutioOutputFile, FileType, FlowRun, FlowRunStatus, flowStructureUtil, isNil, Issue, IssueStatus, ListIssuesParams, PopulatedIssue, SeekPage, spreadIfDefined, StepOutputStatus, TelemetryEventName } from '@activepieces/shared'
 import dayjs from 'dayjs'
 import { FastifyBaseLogger } from 'fastify'
+import { LessThan } from 'typeorm'
 import { repoFactory } from '../../core/db/repo-factory'
+import { fileService } from '../../file/file.service'
 import { buildPaginator } from '../../helper/pagination/build-paginator'
 import { paginationHelper } from '../../helper/pagination/pagination-utils'
 import { telemetry } from '../../helper/telemetry.utils'
 import { flowVersionService } from '../flow-version/flow-version.service'
 import { IssueEntity } from './issues-entity'
+
 const repo = repoFactory(IssueEntity)
 
+const extractFailedStepId = async (flowRun: FlowRun, log: FastifyBaseLogger) => {
+    const flowSteps = await fileService(log).getDataOrThrow({
+        projectId: flowRun.projectId, 
+        fileId: flowRun.logsFileId ?? '', 
+        type: FileType.FLOW_RUN_LOG,
+    })
+    const rawData = Buffer.from(flowSteps.data).toString('utf-8')
+    const executionOutput = JSON.parse(rawData) as ExecutioOutputFile
+    const executionState = executionOutput.executionState
+
+    const failedStep = Object.entries(executionState.steps).find(
+        ([_, step]) => step.status === StepOutputStatus.FAILED,
+    )
+
+    if (!failedStep) {
+        throw new Error('No failed step found')
+    }
+
+    return failedStep[0]
+}
+
+const getStepFromFlow = async (flowId: string, stepId: string, log: FastifyBaseLogger) => {
+    const flowVersion = await flowVersionService(log).getLatestLockedVersionOrThrow(flowId)
+    const allSteps = flowStructureUtil.getAllSteps(flowVersion.trigger)
+    const step = allSteps.find(s => s.name === stepId)
+    
+    if (!step) {
```

Comment:
I think this can be moved under `issueServices` as basically this is helper function and people usually reads code from above to below.

It's confusing to read helper function than understand what is the main purpose of file.